File: backend/DTOs/Responses/ApiResponse.cs
using backend.Enums;

namespace backend.DTOs.Responses
{
    public class ApiResponse<T>
    {
        public bool Success { get; set; } = true;
        public T? Data { get; set; } 
        public ErrorCode ErrorCode { get; set; } = ErrorCode.None;
        public string? Message { get; set; } = string.Empty;
    }
}

File: frontend/domain/def/ApiResponse.ts
import { TNullable } from "@/types";
import { IApiResponse } from "../meta/IApiResponse";
import { ErrorCodeEnum } from "../enum/ErrorCodeEnum";

class ApiResponse<T> implements IApiResponse<T> {
  success: boolean;
  data: TNullable<T>;
  message: string;
  errorCode: ErrorCodeEnum;

  constructor(
    success: boolean,
    data: TNullable<T>,
    message: string,
    errorCode: ErrorCodeEnum,
  ) {
    this.success = success;
    this.data = data;
    this.message = message;
    this.errorCode = errorCode;
  }
}

export { ApiResponse };
